### PR TITLE
Format chat logs for logstash

### PR DESCRIPTION
**Pull request by gitkuba**
[SUS-489](https://wikia-inc.atlassian.net/browse/SUS-489) @macbre @rogatty

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -23,11 +28,28 @@ module.exports.forModule = function (callingModule, filters) {
...
+						'@timestamp': timestamp,
+						'@message': options.message,
+						'@source_host': os.hostname(),
+						'@fields': {
+							'level': options.level,
+							'app_name': 'chat',
+							'data_center': process.env.WIKIA_DATACENTER,
+							'environment': process.env.WIKIA_ENVIRONMENT,
+							'pid': process.pid,
+							'program': process.title
```
Comment (macbre) [inline logger.js]:
```
/home/travis/build/Wikia/chat/logger.js
  42:8  error  Unnecessarily quoted property 'level' found        quote-props
  43:8  error  Unnecessarily quoted property 'app_name' found     quote-props
  44:8  error  Unnecessarily quoted property 'data_center' found  quote-props
  45:8  error  Unnecessarily quoted property 'environment' found  quote-props
  46:8  error  Unnecessarily quoted property 'pid' found          quote-props
  47:8  error  Unnecessarily quoted property 'program' found      quote-props
```

File: logger.js [hunk 2]
```diff
@@ -4,17 +4,22 @@
...
 winston.setLevels(winston.config.syslog.levels);
 winston.add(Syslog, {});
 
 module.exports = {};
 
 module.exports.forModule = function (callingModule, filters) {
 	var parts = callingModule.filename.split('/'),
-		label = parts.pop();
+		label = parts.pop(),
+		timestamp = new Date().toJSON(),
```
Comment (macbre) [inline logger.js]: Let's calculate `timestamp` when returning an object to be sent to syslog.

File: logger.js [hunk 3]
```diff
@@ -23,11 +28,28 @@ module.exports.forModule = function (callingModule, filters) {
...
 			}),
 			new winston.transports.Syslog({
-				label: label,
-				handleExceptions: true
+				handleExceptions: true,
+				formatter: function (options) {
+					entry = {
+						'@timestamp': timestamp,
+						'@message': options.message,
+						'@source_host': os.hostname(),
```
Comment (macbre) [inline logger.js]: We can cache it in a  global variable in module's scope to avoid `uname` system call.

File: logger.js [hunk 4]
```diff
@@ -4,17 +4,22 @@
...
 winston.add(Syslog, {});
 
 module.exports = {};
 
 module.exports.forModule = function (callingModule, filters) {
 	var parts = callingModule.filename.split('/'),
-		label = parts.pop();
+		label = parts.pop(),
+		timestamp = new Date().toJSON(),
+		entry;
```
Comment (rogatty) [inline logger.js]: This could be declared inside `formatter: function (options) { ... }` to limit the scope.

File: logger.js [hunk 5]
```diff
@@ -4,17 +4,22 @@
  * This file contains chat logging class and some socket.io logger related hacks.
  *
  */
+var os = require('os');
 var winston = require('winston');
 var Syslog = require('winston-syslog').Syslog; // eslint-disable-line no-unused-vars
 
+process.title = 'nodejs-chat';
```
Comment (macbre) [inline logger.js]: I'd rather avoid changing state of global `nodejs` objects. Let's simply sent "nodejs-chat" as a value of `program` when pushing messages to syslog (we can use a const for this value).

File: logger.js [hunk 6]
```diff
@@ -4,17 +4,21 @@
...
```
Comment (rogatty) [inline logger.js]: This is probably what gets parsed in https://github.com/Wikia/chef-repo/blob/master/cookbooks/syslogNg/templates/default/syslog-ng-client.conf.erb#L171-L174

File: logger.js (hunk 1 above)
Comment (gitkuba) [inline logger.js]: @macbre ping

File: logger.js [hunk 7]
```diff
@@ -4,8 +4,12 @@
  * This file contains chat logging class and some socket.io logger related hacks.
  *
  */
+var os = require('os');
 var winston = require('winston');
 var Syslog = require('winston-syslog').Syslog; // eslint-disable-line no-unused-vars
+var version = require('./package.json').version;
```
Comment (macbre) [inline logger.js]: :+1:

File: server.js [hunk 8]
```diff
@@ -1048,6 +1048,11 @@ function initSocketLoggerContext(socket) {
 		if (socket.username) {
 			meta.username = socket.username;
 		}
+		if (socket.cityId) {
+			meta.wikiId = socket.cityId;
+		}
+
+		meta.privateRoom = socket.privateRoom;
```
Comment (macbre) [inline server.js]: :+1:

**Comment by mixth-sense** [comment]
Tested and accepted. :medal_sports:

**Pull request merged**